homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -285,43 +295,11 @@ def extra_state_attributes(self):
         if self.vacuum_state is not None:
             attrs.update(
                 {
-                    ATTR_DO_NOT_DISTURB: STATE_ON
-                    if self.dnd_state.enabled
-                    else STATE_OFF,
-                    ATTR_DO_NOT_DISTURB_START: str(self.dnd_state.start),
-                    ATTR_DO_NOT_DISTURB_END: str(self.dnd_state.end),
-                    # Not working --> 'Cleaning mode':
-                    #    STATE_ON if self.vacuum_state.in_cleaning else STATE_OFF,
-                    ATTR_CLEANING_TIME: int(
-                        self.vacuum_state.clean_time.total_seconds() / 60
-                    ),
-                    ATTR_CLEANED_AREA: int(self.vacuum_state.clean_area),
-                    ATTR_CLEANING_COUNT: int(self.clean_history.count),
-                    ATTR_CLEANED_TOTAL_AREA: int(self.clean_history.total_area),
-                    ATTR_CLEANING_TOTAL_TIME: int(
-                        self.clean_history.total_duration.total_seconds() / 60
-                    ),
-                    ATTR_MAIN_BRUSH_LEFT: int(
-                        self.consumable_state.main_brush_left.total_seconds() / 3600
-                    ),
-                    ATTR_SIDE_BRUSH_LEFT: int(
-                        self.consumable_state.side_brush_left.total_seconds() / 3600
-                    ),
-                    ATTR_FILTER_LEFT: int(
-                        self.consumable_state.filter_left.total_seconds() / 3600
-                    ),
-                    ATTR_SENSOR_DIRTY_LEFT: int(
-                        self.consumable_state.sensor_dirty_left.total_seconds() / 3600
-                    ),
                     ATTR_STATUS: str(self.vacuum_state.state),
                     ATTR_MOP_ATTACHED: self.vacuum_state.is_water_box_attached,
                 }
```

Comment:
The mop attached is already a sensor, so that could be removed. Should we expose the error & the state (which is the more fine-grained variant what's the state homeassistant uses as a main state) as sensors, too?

Reply:
yea i was planning indeed to make a switch sensor for error and state

Reply:
But not 100 sure if that is ok, my idea to make it a select sensor is bc you then can use this in automations etc pretty nicely. But then ofc the actual option selector will not work.

But the fact that the state is already translated, do we really need to add it?

Reply:
I suggest not exposing the error as any additional sensors for now. Just follow the entity design for now.

https://developers.home-assistant.io/docs/core/entity/vacuum

Extra state attributes not part of the entity model should be avoided though. Those are better exposed as separate (binary) sensors.

Reply:
Yea based on this, then i do not think status and error should be a sep sensor 🤔. Unless we explicit want to expose a more detailed state via sensor that is specific to xiaomi vacuums. But then imo we should have use cases. e.g if DOCKER != CHARGING, then we can expose a binary charging sensor etc etc.

Reply:
I agree that there's no much reason to expose the status separately (l thought there was a different mode for mopping which might have been useful for someone, but alas, there isn't). I can see some use for the error state (inform about which error happened), but that can be added later if wanted.